### PR TITLE
537 AssemblyScript Counter Contract READMEs

### PR DESCRIPTION
**Pull request by ACStone-MTS**
### Related links

[Document Assembly script based Counter contract #537](https://github.com/casper-network/docs/issues/537)

### Changes

Adding readme files that explain the example contracts related to the "Counter" in AssemblyScript

### CODE REVIEW

**Comment by ACStone-MTS** [comment]
@mpapierski When you get a chance, could you take a look and see if I missed anything glaring here?

File: smart_contracts/contracts_as/example/increment-counter/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  }
+  let value = fromBytesI32(<Uint8Array>countData).unwrap();
+  CL.ret(CLValue.fromI32(value));
+}
+
+```
+
+### Step 3. Defining the `Call` Function
+
+We will also define the `Call` function that will start the code execution and perform the actual code installation. This function initializes the contract by setting the counter to 0 and creates `NamedKeys` in the current context. Specifically, it creates the keys `counter_package_name` and `counter_access_uref`.
```
Comment (darthsiroftardis) [inline smart_contracts/contracts_as/example/increment-counter/README.md]:
```suggestion
We will also define the `Call` function that will start the code execution and perform the contract installation. This function initializes the contract by setting the counter to 0 and creates `NamedKeys` in the current context. Specifically, it creates the keys `counter_package_name` and `counter_access_uref`.
```

Comment (darthsiroftardis) [inline smart_contracts/contracts_as/example/increment-counter/README.md]: Also some changes around the wording of creating NamedKeys, I'd rather we say we create the NamedKeys to pass them along to the contract where they are persisted within the contracts context.

**Review by darthsiroftardis — CHANGES_REQUESTED** [review]
I believe the READMEs have been interchanged, once that correction has been made, I can approve

**Review by darthsiroftardis — APPROVED** [review]

**Review by ipopescu — APPROVED** [review]
Wonderful to have this!

**Pull request merged**